packages/vite/src/node/ssr/ssrModuleLoader.ts

```diff
@@ -46,7 +40,13 @@ export async function ssrLoadModule(
 
   const modulePromise = instantiateModule(url, server, context, urlStack)
   pendingModules.set(url, modulePromise)
-  modulePromise.catch(() => {}).then(() => pendingModules.delete(url))
+  modulePromise
+    .catch(() => {
+      pendingImports.delete(url)
+    })
```

Comment:
This is no longer needed now, no?
```suggestion
```

Reply:
Still needed, since `instantiateModule` promise will reject when a `ssrImport` promise is rejected, which happens after `pendingImports` is set but before it's cleaned up.

Reply:
Oh... sorry for the noise then. I was confused by this comment https://github.com/vitejs/vite/pull/3950#discussion_r680722085, I thought that both bodies of catch and then were equal. Then there isn't a difference in changing the then to finally.

Reply:
Right, just more semantically clear.